pkg/service/node.go

```diff
@@ -122,6 +133,29 @@ func (n *NodeService) NodeUnpublishVolume(ctx context.Context, req *csi.NodeUnpu
 	return &csi.NodeUnpublishVolumeResponse{}, nil
 }
 
+func (n *NodeService) publishBlockVolume(req *csi.NodePublishVolumeRequest, device string) (*csi.NodePublishVolumeResponse, error) {
+	klog.Info("Publishing block volume, device: %s, req: %+v", device, req)
+	file, err := os.OpenFile(req.TargetPath, os.O_CREATE, os.FileMode(0644))
+	defer file.Close()
+	if err != nil {
+		if !os.IsExist(err) {
+			return nil, status.Errorf(codes.Internal, "Failed to create targetPath %s, err: %v", req.TargetPath, err)
+		}
+	}
+
+	mounter := mount.New("")
+	err = mounter.Mount(device, req.TargetPath, "", []string{"bind"})
+	if err != nil {
+		if removeErr := os.Remove(req.TargetPath); removeErr != nil {
+			return nil, status.Errorf(codes.Internal, "Failed to remove mount target %v, err: %v, mount error: %v", req.TargetPath, removeErr,)
```

Comment:
is there a nicer way?

Reply:
your missing the `err` for the ErrorF, that's it I think:
```return nil, status.Errorf(
     codes.Internal,
     "Failed to remove mount target %v, err: %v, mount error: %v",
     req.TargetPath,
     removeErr,
     err)
```

Reply:
```suggestion
			return nil, status.Errorf(codes.Internal, "Failed to remove mount target %v, err: %v, mount error: %v", req.TargetPath, removeErr, err)
```

Reply:
:facepalm:

Reply:
apparently trailing commas are allowed